cli/listener/src/main.rs

```diff
@@ -531,81 +694,168 @@ fn main() {
-        chrono::Utc.ymd(2021, 7, 8).and_hms(0, 0, 0),
-        SOLIDO_ID.to_owned(),
-    )
-    .expect("Failed when getting APY for period");
-    assert_eq!(apy.unwrap().annual_percentage_rate(), 4.7989255185326485);
-}
 
-// When computing the APY, we have to call `growth_factor` which divides two
-// Rational numbers. Previously when dividing two rationals our implementation
-// returned another Rational. In other words, for dividing `a/b` by `c/d`, we
-// did `a*d/b*c`. In this case, `a*d` or `b*c` could overflow, we now return an
-// `f64` instead of a Rational and avoid multiplying two large numbers that
-// could overflow.
-#[test]
-fn test_rationals_do_not_overflow() {
-    use chrono::TimeZone;
-    let conn = Connection::open_in_memory().expect("Failed to open sqlite connection.");
-    create_db(&conn).unwrap();
-    let exchange_rate = ExchangeRate {
-        id: 0,
-        timestamp: chrono::Utc.ymd(2022, 01, 28).and_hms(11, 58, 39),
-        slot: 108837851,
-        epoch: 270,
-        pool: SOLIDO_ID.to_owned(),
-        price_lamports_numerator: 1936245653069130,
-        price_lamports_denominator: 1893971837707973,
-    };
-    insert_price(&conn, &exchange_rate).unwrap();
-
-    let exchange_rate = ExchangeRate {
-        id: 0,
-        timestamp: chrono::Utc.ymd(2022, 02, 28).and_hms(11, 58, 39),
-        slot: 118837851,
-        epoch: 275,
-        pool: SOLIDO_ID.to_owned(),
-        price_lamports_numerator: 1936245653069130,
-        price_lamports_denominator: 1892971837707973,
-    };
-    insert_price(&conn, &exchange_rate).unwrap();
+    #[test]
+    fn test_get_average_apy() {
+        let conn = Connection::open_in_memory().expect("Failed to open sqlite connection.");
+        create_db(&conn).unwrap();
+        let exchange_rate = ExchangeRate {
+            id: 0,
+            timestamp: chrono::Utc.ymd(2020, 8, 8).and_hms(0, 0, 0),
+            slot: 1,
+            epoch: 1,
+            pool: SOLIDO_ID.to_owned(),
+            price_lamports_numerator: 1,
+            price_lamports_denominator: 1,
+        };
+        insert_price(&conn, &exchange_rate).unwrap();
+        let exchange_rate = ExchangeRate {
+            id: 0,
+            timestamp: chrono::Utc.ymd(2021, 1, 8).and_hms(0, 0, 0),
+            slot: 2,
+            epoch: 2,
+            pool: SOLIDO_ID.to_owned(),
+            price_lamports_numerator: 1394458971361025,
+            price_lamports_denominator: 1367327673971744,
+        };
+        insert_price(&conn, &exchange_rate).unwrap();
+        let apy = get_interval_price_for_period(
+            conn.unchecked_transaction().unwrap(),
+            chrono::Utc.ymd(2020, 7, 7).and_hms(0, 0, 0),
+            chrono::Utc.ymd(2021, 7, 8).and_hms(0, 0, 0),
+            SOLIDO_ID.to_owned(),
+        )
+        .expect("Failed when getting APY for period");
+        assert_eq!(apy.unwrap().annual_percentage_rate(), 4.7989255185326485);
```

Comment:
```suggestion
        assert_eq!(apy.unwrap().annual_percentage_yield(), 4.7989255185326485);
```
We should also rename this for consistency, see also my earlier comment.